src/lib/run.rs

```diff
@@ -36,6 +35,49 @@ pub fn run(opts: Opts) -> Result<(), anyhow::Error> {
     // and so all accepted records will go into file(s) for one sample with no Undetermined
     let is_no_demux = samples.len() == 1 && samples[0].barcode.len() == 0;
 
+    // If the input FASTQs are in fact path prefixes, then glob in the FASTQs and create the
+    // read structure based on the kind/kind-number inferred from the file name.
+    let opts = if opts.fastqs.iter().all(|f| f.is_file()) {
+        // do nothing
+        opts
+    } else if opts.fastqs.iter().all(|f| !f.is_file()) {
+        ensure!(
+            opts.read_structures.is_empty(),
+            "Read Structure must not be given when the input FASTQs are a path prefix."
+        );
+
+        // glob all the FASTQs
+        // Important: sort by kind then kind number so the output kind number is ordered correctly
+        let input_fastqs: Vec<InputFastq> = opts
+            .fastqs
+            .iter()
+            .flat_map(|prefix| InputFastq::glob(&prefix).unwrap())
+            .sorted_by(|left, right| {
+                left.kind.cmp(&right.kind).then(left.kind_number.cmp(&right.kind_number))
+            })
+            .collect();
+
+        // build read structures, one per input FASTQ
+        let read_structures: Vec<ReadStructure> =
+            input_fastqs.iter().map(InputFastq::read_structure).collect();
+
+        // extract the list of input FASTQs
+        let fastqs: Vec<PathBuf> =
+            input_fastqs.iter().map(|input_fastq| input_fastq.path.clone()).collect();
+
+        Opts { fastqs, read_structures, ..opts }
+    } else {
+        bail!("Input FASTQS (--fastqs) must either all be files or all path prefixes, not a mixture of both")
```

Comment:
Is it intentional that this code can work with _multiple_ path prefixes?  I don't think it's an issue, but it might be simpler to validate we only get one if it's a prefix?